pkg/envoy/xds_server.go

```diff
@@ -1311,19 +1321,43 @@ func getHTTPRule(secretManager certificatemanager.SecretManager, h *api.PortRule
 						},
 					})
 				} else {
-					// Only header presence needed
+					// Value is empty, but could be we need to set an SDS value for it instead
+					if !inlineSecrets {
+						// Need to add a HeaderMatch so we can specify the SDS value here.
+						log.Debugf("HeaderMatches: Adding %s because SDS value is required", hdr.Name)
+						headerMatches = append(headerMatches, &cilium.HeaderMatch{
+							MismatchAction: mismatch_action,
+							Name:           hdr.Name,
+							ValueSdsSecret: namespacedNametoSyncedSDSSecretName(types.NamespacedName{
+								Namespace: hdr.Secret.Namespace,
+								Name:      hdr.Secret.Name,
+							}, policySecretsNamespace),
+						})
+					}
+					// Check for header presence in headers
```

Comment:
is it correct to add both (`cilium.HeaderMatch` & `HeaderMatcher`) in case the value is empty and `!inlineSecrets`? Or should we handle the `// Check for header presence in headers` in an else condition?

Reply:
Refactored this in 2652aedc2473215bbeaa4b502358d315673ed5d8, what's there is now definitely correct, with tests, and verified with the connectivity tests. Trust me on this, or the rather the six hours I spent testing today.